### ISSUE TITLE
Flaky test csharp.Grpc.Reflection.Tests.ReflectionClientServerTest

### ISSUE THREAD

**Issue by muxi** [grpc/grpc#9958]
Happened in https://grpc-testing.appspot.com/job/gRPC_pull_requests_linux/3130/console

```
NUnitLite 3.2.0 (.NET 4.5)
Copyright (C) 2016 Charlie Poole

Runtime Environment
   OS Version: Unix 4.2.0.42
  CLR Version: 4.0.30319.42000

Test Files
    /var/local/git/grpc/src/csharp/Grpc.Reflection.Tests/bin/Debug/Grpc.Reflection.Tests.exe

Test Filters
    Test: Grpc.Reflection.Tests.ReflectionClientServerTest

D0302 00:38:02.034513 Grpc.Core.Internal.UnmanagedLibrary Attempting to load native library "/var/local/git/grpc/src/csharp/Grpc.Reflection.Tests/bin/Debug/libgrpc_csharp_ext.x64.so"
D0302 00:38:02.072379 Grpc.Core.Internal.NativeExtension gRPC native library loaded successfully.
=> Grpc.Reflection.Tests.ReflectionClientServerTest.FileByFilename
=> Grpc.Reflection.Tests.ReflectionClientServerTest.FileByFilename_NotFound
=> Grpc.Reflection.Tests.ReflectionClientServerTest.FileContainingExtension
=> Grpc.Reflection.Tests.ReflectionClientServerTest.FileContainingSymbol
=> Grpc.Reflection.Tests.ReflectionClientServerTest.FileContainingSymbol_NotFound
=> Grpc.Reflection.Tests.ReflectionClientServerTest.ListServices

Errors and Failures

1) Error : Grpc.Reflection.Tests.ReflectionClientServerTest.FileByFilename
Grpc.Core.RpcException : Status(StatusCode=Unavailable, Detail="{"created":"@1488415082.394837088","description":"Endpoint read failed","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.c","file_line":1851,"grpc_status":14,"occurred_during_write":0,"referenced_errors":[{"created":"@1488415082.394826118","description":"Socket closed","fd":41,"file":"src/core/lib/iomgr/tcp_posix.c","file_line":249,"target_address":"ipv6:[::1]:46068"}]}")
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at Grpc.Core.Internal.ClientResponseStream`2+<MoveNext>c__async0[TRequest,TResponse].MoveNext () [0x0011e] in <12c8ca5e3f4147c692e4d4afb0c7000e>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at Grpc.Reflection.Tests.ReflectionClientServerTest+<SingleRequestAsync>c__async6.MoveNext () [0x000fa] in <750b283b187142849717056505fcf575>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at Grpc.Reflection.Tests.ReflectionClientServerTest+<FileByFilename>c__async1.MoveNext () [0x00078] in <750b283b187142849717056505fcf575>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
  at NUnit.Framework.Internal.ExceptionHelper.Rethrow (System.Exception exception) [0x00006] in <518ab56399554eed9f0def0d872d8d9a>:0 
  at NUnit.Framework.Internal.AsyncInvocationRegion+AsyncTaskInvocationRegion.WaitForPendingOperationsToComplete (System.Object invocationResult) [0x00030] in <518ab56399554eed9f0def0d872d8d9a>:0 
  at NUnit.Framework.Internal.Commands.TestMethodCommand.RunAsyncTestMethod (NUnit.Framework.Internal.TestExecutionContext context) [0x00038] in <518ab56399554eed9f0def0d872d8d9a>:0 

Run Settings
    Number of Test Workers: 1
    Work Directory: /var/local/git/grpc
    Internal Trace: Off

Test Run Summary
  Overall result: Failed
  Test Count: 6, Passed: 5, Failed: 1, Inconclusive: 0, Skipped: 0
    Failed Tests - Failures: 0, Errors: 1, Invalid: 0
  Start time: 2017-03-02 00:38:01Z
    End time: 2017-03-02 00:38:02Z
    Duration: 0.737 seconds


FAILED: csharp.Grpc.Reflection.Tests.ReflectionClientServerTest [ret=1, pid=6605]
```

**Comment by dgquintas** [comment]
Seen again: https://grpc-testing.appspot.com/job/gRPC_pull_requests_linux/4663/testReport/junit/(root)/csharp_linux_dbg_native/csharp_Grpc_Reflection_Tests_ReflectionClientServerTest/

**Comment by jtattermusch** [comment]
Likely to be fixed by #10756
